cli/module_graph2.rs

```diff
@@ -469,10 +568,11 @@ pub struct TranspileOptions {
 #[derive(Debug)]
 pub struct Graph2 {
   handler: Rc<RefCell<dyn SpecifierHandler>>,
-  maybe_ts_build_info: Option<String>,
+  maybe_tsbuildinfo: Option<String>,
   modules: HashMap<ModuleSpecifier, Module>,
   redirects: HashMap<ModuleSpecifier, ModuleSpecifier>,
   roots: Vec<ModuleSpecifier>,
+  roots_dynamic: bool,
```

Comment:
What does this field do? Could you add a comment?

Reply:
Specifically it deals with the situation where a dependency has already been loaded and compiled during a run, but then is subsequently loaded as a dynamic import when the `--reload` flag is passed... If in that case the dynamic import has a valid emit, then it will not be reloaded.

As I mentioned in another conversation, I don't think our current behaviour around `--reload` makes sense, when we can be 100% confident that emit will be reproduced now, but we have integration tests that expect that `--reload` will compile a module, unless it has already been compiled (see: `single_compile_with_reload`).

I will add a comment.

Reply:
I've also opened #8066